lib/python/Tools/MultiBoot.py

```diff
@@ -202,16 +203,17 @@ def saveKernel(bootSlots, slotCode, kernel):
 							if "rootsubdir" in line:
 								bootSlots[slotCode]["kernel"] = self.getParam(line, "kernel")
 								bootSlots[slotCode]["rootsubdir"] = self.getParam(line, "rootsubdir")
-							elif "sda" in line:
-								saveKernel(bootSlots, slotCode, "/dev/sda%s" % line.split("sda", 1)[1].split(" ", 1)[0])
+							elif match("sd[abcd]", line):
```

Comment:
Rather than using a regular expression, would a line.startswith((sda, sdb, sdc, sdd)) be faster?

Reply:
Thanks for looking into it

in the future there is also a small fix in file /etc/udev/mount-helper.sh
Blacklist boot device
# detected multiboot sdcard
                        if [ $stbcheck == "viper4k" ] || [ $stbcheck == "sf8008" ] || [ $stbcheck == "sf8008m" ] || [ $stbcheck == "ustym4kpro" ] || [ $stbcheck == "beyonwizv2" ] || [ $stbcheck == "gbmv200" ]; then
                                DEVCHECK=`expr substr $MDEV 1 3`
                                if [ "${DEVCHECK}" == "sda" ] ; then
                                        DEVSIZE=`cat /sys/block/sda/sda1/size`
                                        if [ $DEVSIZE -lt "32769" ]; then
                                                BLACKLISTED=`echo ${BLACKLISTED} sda`
                                        fi
                                fi